visualization/app/codeCharta/util/fileValidator.ts

```diff
@@ -1,15 +1,80 @@
 import { CodeMapNode } from "../codeCharta.model"
+import { ValidationError, Validator, ValidatorResult } from "jsonschema"
+
+const jsonSchema = require("./schema.json")
+const latestApiVersion = require("../../../package.json").codecharta.apiVersion
+
+interface ApiVersion {
+	major: number
+	minor: number
+}
+
+export interface CCValidationResult {
+	error: string[]
+	warning: string[]
+	title: string
+}
 
 export class FileValidator {
-	private static hasUniqueChildren(node: CodeMapNode) {
+	private static FILE_IS_INVALID = ["file is empty or invalid", "Error Loading File"]
```

Comment:
I think having nice objects with readable properties is nicer: Something like this:
```
private static FILE_IS_INVALID = { message: "file is empty or invalid", title: "Error Loading File" }
```

Or what might be also nice, is one object like below:
```
private static ERROR_MESSAGES = {
  fileIsInvalid: {
    title: "some title",
    message: "some message"
  },
  ...
}
```

Reply:
And I think titles that are similar to each other, are easier to recognise as titles. Like always starting with `Error: abc`/`Warning: abc` or always ending with `ABC Error` / `ABC Warning`

Reply:
I really like the second idea, makes it much more readable
Should we export it as a constant then?

Reply:
cool :) nah, if we don't move the object into a separate file, we don't need to export it